src/element/textElement.ts

```diff
@@ -890,18 +899,10 @@ export const computeContainerDimensionForBoundText = (
   return dimension + padding;
 };
 
-export const getMaxContainerWidth = (container: ExcalidrawElement) => {
+export const getBoundTextMaxWidth = (container: ExcalidrawElement) => {
   const width = getContainerDims(container).width;
   if (isArrowElement(container)) {
-    const containerWidth = width - BOUND_TEXT_PADDING * 8 * 2;
-    if (containerWidth <= 0) {
-      const boundText = getBoundTextElement(container);
-      if (boundText) {
-        return boundText.width;
-      }
-      return BOUND_TEXT_PADDING * 8 * 2;
-    }
-    return containerWidth;
```

Comment:
why was this needed and is not needed anymore?

Reply:
This was never needed actually coz if you notice, `getContainerDims` will never return dimensions <=0 for arrow the min width returned is `300` so this code never gets executed.
So I believe only when that check for 300 wasn't there only then this was being used otherwise not.